doc/reference/usermode/syscalls.rst

```diff
@@ -331,8 +332,198 @@ For example:
     }
     #include <syscalls/k_sem_take_mrsh.c>
 
-Verification Policies
-=====================
+
+Verification Memory Access Policies
+===================================
+
+Parameters passed to system calls by reference require special handling,
+because the value of these parameters can be changed at any time by any
+user thread that has access to the memory that parameter points to. If the
+kernel makes any logical decisions based on the contents of this memory, this
+can open up the kernel to attacks even if checking is done. This is a class
+of exploits known as TOCTOU (Time Of Check to Time Of Use).
+
+The proper procedure to mitigate these attacks is to make a copies in the
+verification function, and only perform parameter checks on the copies, which
+user threads will never have access to. The implementation functions get passed
+the copy and not the original data sent by the user. The
+:c:func:`z_user_to_copy()` and :c:func:`z_user_from_copy()` APIs exist for
```

Comment:
These names are kind of confusing. Any reason to not use `z_copy_to_user` and `z_copy_from_user`?

Reply:
these APIs already exist, and Zephyr naming convention (not followed 100%) is to have the verb last.